terminal/bundles/org.eclipse.terminal.view.ui/src/org/eclipse/terminal/view/ui/internal/ConsoleManager.java

```diff
@@ -396,21 +380,21 @@ public CTabItem openConsole(TerminalViewId tvid, String title, String encoding,
 			}
 
 			// Create a new tab item
-			item = manager.createTabItem(title, encoding, connector, data, flags);
+			item = Optional.ofNullable(manager.createTabItem(title, encoding, connector, data, flags));
 		}
 		// If still null, something went wrong
-		if (item == null) {
-			return null;
+		if (item.isEmpty()) {
+			throw cannotCreateConsole();
 		}
 
 		// Make the item the active console
-		manager.bringToTop(item);
+		item.filter(CTabItem.class::isInstance).map(CTabItem.class::cast).ifPresent(manager::bringToTop);
 
 		// Make sure the terminals view has the focus after opening a new terminal
 		view.setFocus();
 
 		// Return the tab item of the opened console
-		return item;
+		return item.get();
```

Comment:
if you use  `item.orThrow(...)` here we can avoid the check in line 386/387

Reply:
done